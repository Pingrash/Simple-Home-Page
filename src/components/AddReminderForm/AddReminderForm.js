import React, { useState } from 'react';
import PropTypes from 'prop-types';

import Paper from '@material-ui/core/Paper';
import FormControl from '@material-ui/core/FormControl';
import Button from '@material-ui/core/Button';
import Input from '@material-ui/core/Input';
import InputLabel from '@material-ui/core/InputLabel';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(theme => ({
  container: {
    marginTop: '1rem',
    padding: '1rem'
  },
  form: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column'
  },
  button: {
    marginTop: '1rem'
  }
}));

const AddReminderForm = ({ addReminder }) => {
  const [title, setTitle] = useState('');
  const [body, setBody] = useState('');

  const classes = useStyles();

  const clearFields = () => {
    setTitle('');
    setBody('');
  };

  return (
    <Paper className={classes.container}>
      <Typography component='h6' variant='h6'>
        Add Reminder
      </Typography>
      <form
        onSubmit={e => {
          e.preventDefault();
          const newReminder = {
            title: title,
            body: body
          };
          addReminder(newReminder);
          clearFields();
        }}
        className={classes.form}
      >
        <FormControl>
          <InputLabel htmlFor='title' color='primary' required>
            Title:
          </InputLabel>
          <Input
            id='title'
            type='text'
            onChange={e => setTitle(e.target.value)}
            fullWidth
            data-test='form-field'
            value={title}
          />
        </FormControl>
        <FormControl>
          <InputLabel htmlFor='body' color='primary' required>
            Body:
          </InputLabel>
          <Input
            id='body'
            type='text'
            onChange={e => setBody(e.target.value)}
            fullWidth
            data-test='form-field'
            value={body}
          />
        </FormControl>
        <Button
          type='submit'
          color='primary'
          fullWidth
          variant='contained'
          className={classes.button}
          data-test='submit-button'
        >
          Submit
        </Button>
      </form>
    </Paper>
  );
};

AddReminderForm.propTypes = {
  addReminder: PropTypes.func.isRequired
};

export default AddReminderForm;
