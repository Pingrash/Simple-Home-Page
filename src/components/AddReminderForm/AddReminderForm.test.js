import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import AddReminderForm from './AddReminderForm';
import {
  checkProps,
  testFunc,
  findByDataTest
} from '../../../Utilities';

const props = { addReminder: testFunc };

const setup = () => {
  const component = shallow(<AddReminderForm {...props} />);
  return component;
};

describe('Snapshot', () => {
  it('Should match snapshot', () => {
    const tree = renderer
      .create(<AddReminderForm {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('AddReminderForm renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should have two inputs', () => {
    const inputs = findByDataTest(wrapper, 'form-field');
    expect(inputs.length).toBe(2);
  });

  it('Should have a submit button', () => {
    const button = findByDataTest(wrapper, 'submit-button');
    expect(button).toBeTruthy();
  });
});

describe('AddReminderForm should have correct props', () => {
  it('Should not return a prop error', () => {
    const propsError = checkProps(AddReminderForm, props);
    expect(propsError).toBeUndefined();
  });
});
