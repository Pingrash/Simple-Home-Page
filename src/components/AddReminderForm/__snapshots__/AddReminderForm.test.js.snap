// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should match snapshot 1`] = `
<div
  className="MuiPaper-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    Add Reminder
  </h6>
  <form
    className="makeStyles-form-2"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="title"
      >
        Title:
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        data-test="form-field"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="title"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
        data-shrink={false}
        htmlFor="body"
      >
        Body:
        <span
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
        data-test="form-field"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          className="MuiInputBase-input MuiInput-input"
          disabled={false}
          id="body"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-3 MuiButton-containedPrimary MuiButton-fullWidth"
      data-test="submit-button"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Submit
      </span>
    </button>
  </form>
</div>
`;
